lib/cache/PackFileCacheStrategy.js

```diff
@@ -784,6 +824,40 @@ class PackFileCacheStrategy {
 			});
 	}
 
+	contextifyIdentifier(ident) {
+		if (!this.contextify) return ident;
+
+		let contextified = this.identCache.get(ident);
+
+		if (contextified) return contextified;
+
+		let matched = false;
+		contextified = ident.replace(
+			ABSOLUTE_PATH_IDENTIFIER_REGEX,
+			(match, path) => {
+				matched = true;
+
+				return "/" + this.contextifyAbsolutePath(path);
+			}
+		);
+
+		if (matched) {
+			this.identCache.set(ident, contextified);
+			return contextified;
+		}
+
+		contextified = ident.replace(RESOLVE_IDENTIFIER_REGEX, (match, path, i) => {
+			return (
+				"path=" +
+				this.contextifyAbsolutePath(path) +
+				ident.slice(i + path.length + 5 /* "path=".length */)
+			);
+		});
+
+		this.identCache.set(ident, contextified);
+		return contextified;
+	}
```

Comment:
Too hacky. `contextify` should be used and cache identifiers should be refactored to use `|` as separator.

Move the logic into `Pack` (.serialize)